### ISSUE TITLE
Test score is inf

### ISSUE THREAD

**Issue by akshathaarodi** [sattree/gap#17]
Hi,

I am able to set up and train the model successfully, but the test scores are inf and the best model is not picked up. How can I fix this one?
<img width="639" alt="Screen Shot 2021-10-12 at 10 57 49 PM copy" src="https://user-images.githubusercontent.com/34102425/137073635-6ae7be22-fe69-4b1c-be8c-8bd8361ea597.png">

EDIT: I suspect it is because of early stopping. But Trn 4 has a better F1 score on val set.

**Comment by sattree** [comment]
@akshathaarodi Hi Akshath, is this still an issue?

**Comment by akshathaarodi** [comment]
Yes!

**Comment by sattree** [comment]
inf is probably just the default value showing up. Wondering why evaluation is being bypassed. Some of the conditionals are likely evaluating to false.

What patience and epoch values are you using? If patience is 3, then why did training terminate after 5 epochs?

**Comment by akshathaarodi** [comment]
I am using default values. I reduced the train_batch_size to help run on a single GPU.


                train_batch_size=3,
                eval_batch_size=32,
                learning_rate=4e-6,
                num_train_epochs=20,
                patience=3

Not sure how to debug this one.

**Comment by sattree** [comment]
So, the execution looks fine. Val score was lowest for epoch #1 and training terminated 3 epochs after that. The final scores are correctly based on the checkpoint corresponding to epoch #1. It's been a while but I'll take a look to see why infs are showing up.

As for you original question, the termination criteria is based on val 'score'. 'score' here refers to log loss which was the primary metric of performance. F1 scores were computed to gain additional insights.

Does this answer your question? Are there any other problems you see?

**Comment by akshathaarodi** [comment]
I see. I was looking at F1 score. Yes, thank you very much!
